kotlin/flatbuffers-kotlin/src/commonMain/kotlin/com/google/flatbuffers/kotlin/JSON.kt

```diff
@@ -0,0 +1,788 @@
+
+private inline fun ReadWriteBuffer.jsonEscape(data: ReadBuffer, start: Int, size: Int) {
+  val replacements = JSON_ESCAPE_CHARS
+  put(LETTER_DOUBLE_QUOTE)
+  var last = start
+  val length: Int = size
+  val ary = data.data()
+  for (i in start until start + length) {
+    val c = ary[i].toUByte()
+    var replacement: ByteArray?
+    if (c.toInt() < 128) {
+      replacement = replacements[c.toInt()]
+      if (replacement == null) {
+        continue
+      }
+    } else {
+      continue
+    }
+    if (last < i) {
+      put(ary, last, i - last)
+    }
+    put(replacement, 0, replacement.size)
+    last = i + 1
+  }
+  if (last < (last + length)) {
+    put(ary, last, (start + length) - last)
+  }
+  put(LETTER_DOUBLE_QUOTE)
+}
+
+// Following escape strategy defined in RFC7159.
+private val JSON_ESCAPE_CHARS: Array<ByteArray?> = arrayOfNulls<ByteArray>(128).apply {
+  this['\n'.toInt()] = "\\n".encodeToByteArray()
+  this['\t'.toInt()] = "\\t".encodeToByteArray()
+  this['\r'.toInt()] = "\\r".encodeToByteArray()
+  this['\b'.toInt()] = "\\b".encodeToByteArray()
+  this[0x0c] = "\\f".encodeToByteArray()
+  this['"'.toInt()] = "\\\"".encodeToByteArray()
+  this['\\'.toInt()] = "\\\\".encodeToByteArray()
+  for (i in 0..0x1f) {
+    this[i] = "\\u${i.toPaddedHex()}".encodeToByteArray()
+  }
+}
+
+private const val LETTER_NEWLINE = '\n'.toByte()
+private const val LETTER_OPEN_OBJECT = '{'.toByte()
+private const val LETTER_CLOSE_OBJECT = '}'.toByte()
+private const val LETTER_OPEN_ARRAY = '['.toByte()
+private const val LETTER_CLOSE_ARRAY = ']'.toByte()
+private const val LETTER_DOUBLE_QUOTE = '"'.toByte()
+private const val LETTER_SINGLE_QUOTE = '\''.toByte()
+private const val LETTER_BACKSLASH = '\\'.toByte()
+private const val LETTER_FORWARDSLASH = '/'.toByte()
+private const val LETTER_f = 'f'.toByte()
+private const val LETTER_a = 'a'.toByte()
+private const val LETTER_r = 'r'.toByte()
+private const val LETTER_t = 't'.toByte()
+private const val LETTER_n = 'n'.toByte()
+private const val LETTER_b = 'b'.toByte()
+private const val LETTER_e = 'e'.toByte()
+private const val LETTER_E = 'E'.toByte()
+private const val LETTER_u = 'u'.toByte()
+private const val LETTER_A = 'A'.toByte()
+private const val LETTER_F = 'F'.toByte()
+private const val LETTER_EOF = (-1).toByte()
+private const val LETTER_COMMA = ','.toByte()
+private const val LETTER_0 = '0'.toByte()
+private const val LETTER_1 = '1'.toByte()
+private const val LETTER_2 = '2'.toByte()
+private const val LETTER_3 = '3'.toByte()
+private const val LETTER_4 = '4'.toByte()
+private const val LETTER_5 = '5'.toByte()
+private const val LETTER_6 = '6'.toByte()
+private const val LETTER_7 = '7'.toByte()
+private const val LETTER_8 = '8'.toByte()
+private const val LETTER_9 = '9'.toByte()
+private const val LETTER_MINUS = '-'.toByte()
+private const val LETTER_PLUS = '+'.toByte()
+private const val LETTER_DOT = '.'.toByte()
+
+private val doubleConstants = doubleArrayOf(
```

Comment:
wait what? we need this whole big ugly-ass table to be able to parse floats in decimals? Surely that can be done more elegantly..

Reply:
It is actually common to have the power table, like [here](https://github.com/fastfloat/fast_float/blob/main/include/fast_float/fast_table.h#L34). But I guess it can be take out with little loss.

Reply:
Got rid of this